### PR TITLE
[Lottie Exporter] Add support for radial gradient layer

### PR DESCRIPTION
**Pull request by saurabh-ramola**
@AnishGG , Radial Gradient layer is now complete. Can you please test it and let me know if it is correct?

### CODE REVIEW

**Comment by saurabh-ramola** [comment]
I think this will close #934 with respect to support available in Lottie? @AnishGG

**Comment by AnishGG** [comment]
I am testing it now.


> I think this will close #934 with respect to support available in Lottie? @AnishGG

Yes, if this works perfectly, then gradient layers will be supported as of now because there is no support for conical or other gradient layers in lottie format for now.

**Comment by AnishGG** [comment]
@saurabh-ramola I would suggest you to add 1 or 2 example sif files for both gradient layers as well as the convert methods at https://www.gitlab.com/synfig/synfig-tests for validation. ^__^

**Comment by saurabh-ramola** [comment]
@AnishGG I have added the documentation and the test-files ^__^

File: synfig-studio/plugins/lottie-exporter/shapes/gFill.py [hunk 1]
```diff
@@ -12,9 +12,83 @@
 from common.Gradient import Gradient
 sys.path.append("..")
 
+
+def gen_radial_gradient(lottie, layer, idx):
+    """
+    Generates the dictionary correspnding to shapes/gFill.json but with radial gradient
```
Comment (ice0) [inline synfig-studio/plugins/lottie-exporter/shapes/gFill.py:18]: corresponding

File: synfig-studio/plugins/lottie-exporter/shapes/gFill.py [hunk 2]
```diff
@@ -12,9 +12,83 @@
...
+    lottie["nm"] = layer.get_description()
+    lottie["t"] = 2    # 2 means radial gradient layer
+    lottie["s"] = {}   # Starting point of gradient
+    lottie["e"] = {}   # Ending point of gradient
+    lottie["g"] = {}   # Gradient information is stored here
+
+    # Color Opacity
+    opacity = layer.get_param("amount").get()
+    is_animate = is_animated(opacity[0])
+    if is_animate == 2:
```
Comment (ice0) [inline synfig-studio/plugins/lottie-exporter/shapes/gFill.py:35]: Maybe change "2" to constant? Just for a better understanding of what is happening.

File: synfig-studio/plugins/lottie-exporter/shapes/gFill.py [hunk 3]
```diff
@@ -12,9 +12,83 @@
...
+        # Telling the function that this is for opacity
+        opacity[0].attrib['type'] = 'opacity'
+        gen_value_Keyframed(lottie["o"], opacity[0], index.inc())
+
+    else:
+        if is_animate == 0:
```
Comment (ice0) [inline synfig-studio/plugins/lottie-exporter/shapes/gFill.py:41]: Here too (constant)

File: synfig-studio/plugins/lottie-exporter/shapes/gFill.py (hunk 2 above)
Comment (AnishGG) [reply synfig-studio/plugins/lottie-exporter/shapes/gFill.py:35]: Actually, this "2" constant has been used by me at many more places. Like here: 

- https://github.com/synfig/synfig/blob/4c519a9539905788b66101b4b440880949733a6b/synfig-studio/plugins/lottie-exporter/shapes/star.py#L53

- https://github.com/synfig/synfig/blob/4c519a9539905788b66101b4b440880949733a6b/synfig-studio/plugins/lottie-exporter/shapes/star.py#L72
- https://github.com/synfig/synfig/blob/4c519a9539905788b66101b4b440880949733a6b/synfig-studio/plugins/lottie-exporter/shapes/rectangle.py#L34
- And many more places.

@ice0 , may be we can have a seperate PR to deal with this? ^__^

File: synfig-studio/plugins/lottie-exporter/shapes/gFill.py (hunk 3 above)
Comment (AnishGG) [reply synfig-studio/plugins/lottie-exporter/shapes/gFill.py:41]: same thing here also, as mentioned in my above comment

File: synfig-studio/plugins/lottie-exporter/shapes/gFill.py (hunk 2 above)
Comment (ice0) [reply synfig-studio/plugins/lottie-exporter/shapes/gFill.py:35]: Sure.

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**